File: python3/adivinhacao.py
import random

print("************************************")
print("* Bem vindo ao jogo de adivinhacao *")
print("************************************")

numero_secreto = random.randrange(1, 100)
total_tentativas = 0
pontos = 1000

print("Qual o nível de dificuldade?")
print("(1) Fácil (2) Médio (3) Difícil")

nivel = int(input("Defina o nível: "))

if nivel == 1:
    total_tentativas = 20
elif nivel == 2:
    total_tentativas = 10
else:
    total_tentativas = 5

for rodada in range(1, total_tentativas + 1):
    print("Tentativa {} de {}".format(rodada, total_tentativas))
    chute = input("Digite um número de 1 a 100: ")

    chute = int(chute)
    if (chute < 1 or chute > 100):
        print("Você deve digitar um número entre 1 e 100!")
        continue

    acertou = numero_secreto == chute
    menor = numero_secreto < chute
    maior = numero_secreto > chute
    if acertou:
        print("Você acertou e fez {} pontos!".format(pontos))
        break
    else:
        if menor:
            print("Você errou! O número é menor")
        elif maior:
            print("Você errou! O número é maior")
        pontos_perdidos = round(abs(numero_secreto - chute) / 3)
        pontos = pontos - pontos_perdidos

print("Fim do jogo")

File: java2/primeiroprograma/src/Adivinhacao.java
import java.util.Random;
import java.util.Scanner;

public class Adivinhacao {
    public static void main(String[] args) {
        int aleatorio = new Random().nextInt(100);

        Scanner leitura = new Scanner(System.in);
        int resposta = 0;
        int tentativas = 0;

        System.out.println("Digite um número de 0 a 100:");
        while (tentativas < 5) {
            resposta = leitura.nextInt();
            if (resposta == aleatorio) {
                System.out.println("Parabéns, você acertou!");
                break;
            } else {
                System.out.println("Errou! Restam " + (4 - tentativas) + " tentativas");
            }
            tentativas++;
        }
        System.out.println("Fim do jogo");
    }
}
